googleapis_auth/lib/googleapis_auth.dart

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Contains common libraries used across the package.
+///
+/// In most cases, you'll want to import any use either
+/// `auth_io.dart` or `auth_browser.dart` depending on your platform.
+library googleapis_auth;
+
+export 'src/access_credentials.dart';
+export 'src/access_token.dart';
+export 'src/auth_client.dart';
+export 'src/auth_functions.dart';
+export 'src/client_id.dart';
+export 'src/exceptions.dart';
+export 'src/response_type.dart';
+export 'src/service_account_credentials.dart';
```

Comment:
Could we not expose `clientViaServiceAccount` and `clientViaApiKey` using conditional imports/exports?

I'm not sure of there is anything like ADC for js.. but that could also be awesome.

Reply:
I'm confused. Explain?

Reply:
Both `lib/auth_browser.dart` and `lib/auth_io.dart` exports `clientViaApiKey`, so why not export it from `lib/googleapis_auth.dart` with a conditional export..

Reply:
https://github.com/dart-lang/googleapis/issues/198